app/controllers/AddABeneficiaryController.scala

```diff
@@ -0,0 +1,125 @@
+                                       yesNoFormProvider: YesNoFormProvider,
+                                       val controllerComponents: MessagesControllerComponents,
+                                       addAnotherView: AddABeneficiaryView,
+                                       yesNoView: AddABeneficiaryYesNoView,
+                                       val appConfig: FrontendAppConfig
+                                     )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with Enumerable.Implicits {
+
+  val addAnotherForm : Form[AddABeneficiary] = addAnotherFormProvider()
+
+  val yesNoForm: Form[Boolean] = yesNoFormProvider.withPrefix("addABeneficiaryYesNo")
+
+  private def returnToStart(userAffinityGroup : AffinityGroup): Result = userAffinityGroup match {
+    case Agent => Redirect(appConfig.maintainATrustAgentDeclarationUrl)
+    case _ => Redirect(appConfig.maintainATrustIndividualDeclarationUrl)
+  }
+
+  def onPageLoad(): Action[AnyContent] = standardActionSets.verifiedForUtr.async {
+    implicit request =>
+
+      trust.getBeneficiaries(request.userAnswers.utr) map {
+        case Beneficiary(Nil) =>
+          Ok(yesNoView(yesNoForm))
+        case all: Beneficiary =>
+
+          val beneficiaries = new AddABeneficiaryViewHelper(all).rows
+
+          Ok(addAnotherView(
+            form = addAnotherForm,
+            inProgressBeneficiaries = beneficiaries.inProgress,
+            completeBeneficiaries = beneficiaries.complete,
+            heading = all.addToHeading
+          ))
+      }
+  }
+
+  def submitOne(): Action[AnyContent] = standardActionSets.identifiedUserWithData {
+    implicit request =>
+      yesNoForm.bindFromRequest().fold(
+        (formWithErrors: Form[_]) => {
+          BadRequest(yesNoView(formWithErrors))
+        },
+        addNow => {
+          if (addNow) {
+            Redirect(controllers.routes.AddABeneficiaryController.onPageLoad())
+          } else {
+            returnToStart(request.user.affinityGroup)
+          }
+        }
+      )
+  }
+
+  def submitAnother(): Action[AnyContent] = standardActionSets.identifiedUserWithData.async {
+    implicit request =>
+
+      trust.getBeneficiaries(request.userAnswers.utr).map { beneficiaries =>
+        addAnotherForm.bindFromRequest().fold(
+          (formWithErrors: Form[_]) => {
+
+            val rows = new AddABeneficiaryViewHelper(beneficiaries).rows
+
+            BadRequest(
+              addAnotherView(
+                formWithErrors,
+                rows.inProgress,
+                rows.complete,
+                beneficiaries.addToHeading
+              )
+            )
+          },
+          {
+            case AddABeneficiary.YesNow =>
+              Redirect(controllers.routes.AddABeneficiaryController.onPageLoad())
+            case AddABeneficiary.YesLater =>
+              returnToStart(request.user.affinityGroup)
+            case AddABeneficiary.NoComplete =>
```

Comment:
@JoshuaH01 This should go to the task list /overview on the overview and do the POST as per the trustees service.